pkg/client/access_control_records_object_set.go

```diff
@@ -24,6 +27,14 @@ type AccessControlRecordObjectSet struct {
 func (objectSet *AccessControlRecordObjectSet) CreateObject(payload *nimbleos.AccessControlRecord) (*nimbleos.AccessControlRecord, error) {
 	resp, err := objectSet.Client.Post(accessControlRecordPath, payload, &nimbleos.AccessControlRecord{})
 	if err != nil {
+		//process http code 202
+		if strings.Contains(err.Error(), "status (202)") {
+			if resp != nil {
+				ID := resp.(string)
+				// Get object
+				return objectSet.GetObject(ID)
+			}
+		}
```

Comment:
We should not be repeating this logic in every ObjectSet.  Please push the job handling down into the GroupMgmtClient.

Reply:
This cannot be done as the objectSet type is known only to object set layer . Example : ID could be of volume or it could be of snapshot .

Reply:
Yes, this can be done.  The Post function takes the expected response type as an interface:

`func (client *GroupMgmtClient) Post(path string, payload interface{}, respHolder interface{})`

So why not check the http response status within that function?  If you have a 202, then you call Get from the client after the job completes:

`func (client *GroupMgmtClient) Get(path string, id string, intf interface{})`

Reply:
I see . Yes , it seems possible, the only concern is "path" even that is the same for both POST/PUT and GET . thanks @rgcostea